### DISCUSSION TITLE
[Feature Request] Network: Add support for bits speed rate display

### DISCUSSION THREAD

**Discussion by scorpius2k1** [polybar/polybar#2894] [Ideas]
Greetings,

Please excuse me if I missed a setting somewhere in https://github.com/polybar/polybar/wiki/Module:-network, if this is already possible I would appreciate if someone would let me know what option I need to set. Thanks!

**Synopsis:**
Currently, the network module seems to only have an option to output Bytes instead of Bytes OR Bits. For example, being able to display in bits would be useful for seeing your actual speeds compared to what you would expect to see from your ISP and/or just seeing data transfer rates in what is standard "bits" over the network. Speedtest.net is another example use case where the output of Polybar's network module will show "conflicting" results (SpeedTest is in bits, where PB is in bytes). 

Ex:
Speed in MegaBytes/s (seems to be currently the only option in Polybar): 15MB/s
Speed in MegaBits/s (feature request): 120Mb/s

From my understanding, a new "bits" setting if enabled, should just be simple math multiplying the current "byte" option by 8 (as far as I am aware). I believe `speedrate` might be the variable in function `string network::format_speedrate` for devs to modify if the "bits" setting would be enabled, for example, a simple conditional operator to calculate the bits OR byte amount. A new variable, `bUseBits` could be used to used to retrieve the boolean value from a config setting.

`
float speedrate = bUseBits ? 8 * (bytes_diff / time_diff) : bytes_diff / time_diff;
`

https://github.com/polybar/polybar/blob/fc423c092143464062f77b83f78328e8ca96694a/src/adapters/net.cpp#L348

Another link, should it be useful to devs, explains bits/bytes calculations:
https://www.techtarget.com/searchnetworking/definition/Mbps

Thanks for reading and for Polybar!

**Comment by l4n1skyy** [comment]
mbps pls!
